crates/ruff_linter/resources/test/fixtures/pyupgrade/UP008.py

```diff
@@ -141,3 +141,238 @@ def method2(self):
     
     def method3(self):
         super(ExampleWithKeywords, self).some_method()  # Should be fixed - no keywords
+
+# See: https://github.com/astral-sh/ruff/issues/19357
+# Must be detected
+class ParentD1:
+    def f(self):
+        print("!")
+
+class ChildD1(ParentD1):
+    def f(self):
+        if False: __class__ # Python injects __class__ into scope
+        builtins.super(ChildD1, self).f()
+
+ChildD1().f()
```

Comment:
I think we could strip these examples down a little bit, just to make it clearer which parts are important. For example, I don't think the parent class even needs to exist actually, and the same with the following calls. Basically I'd just use this:


```suggestion
class ChildD1:
    def f(self):
        if False: __class__ # Python injects __class__ into scope
        builtins.super(ChildD1, self).f()
```

unless I'm missing some other critical part of the example.

Or we could maybe preserve one of the parent classes and the calls, if it's helpful to be able to run the code. I just don't think we need one parent class per example, if they're all the same, at least.